src/main/java/picard/vcf/FixVcfHeader.java

```diff
@@ -0,0 +1,225 @@
+import htsjdk.variant.vcf.VCFHeaderLine;
+import htsjdk.variant.vcf.VCFHeaderLineCount;
+import htsjdk.variant.vcf.VCFHeaderLineType;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import htsjdk.variant.vcf.VCFStandardHeaderLines;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.CommandLineProgramProperties;
+import picard.cmdline.Option;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.VcfOrBcf;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Tool for replacing or fixing up a VCF header.
+ *
+ * @author Nils Homer
+ */
+@CommandLineProgramProperties(
+        usage = FixVcfHeader.USAGE_SUMMARY + FixVcfHeader.USAGE_DETAILS,
+        usageShort = FixVcfHeader.USAGE_SUMMARY,
+        programGroup = VcfOrBcf.class
+)
+public class FixVcfHeader extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Replaces or fixes a VCF header.";
+    static final String USAGE_DETAILS =
+            "This tool will either replace the header in the input VCF file (INPUT) with the given VCF header (HEADER) or will try to fixup " +
+            "the input header by examining the variants in the input VCF file (INPUT).  In the latter case, this tool will perform two passes " +
+            "over the input VCF, and any INFO and FORMAT fields found in the VCF records but not found in the input VCF header will be added " +
+            "to the output VCF header with dummy descriptions.<br />" +
+            "<h4>Replace header usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar FixVcfHeader \\<br />" +
+            "     I=input.vcf \\<br />" +
+            "     O=fixed.vcf \\<br />" +
+            "     FIX_HEADER=false \\<br />" +
+            "     HEADER=header.vcf" +
+            "</pre>" +
+            "<h4>Fix up usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar FixVcfHeader \\<br />" +
+            "     I=input.vcf \\<br />" +
+            "     O=fixed.vcf \\<br />" +
+            "     FIX_HEADER=true \\<br />" +
+            "</pre>" +
+            "<hr />";
+    @Option(shortName= StandardOptionDefinitions.INPUT_SHORT_NAME, doc="The input VCF/BCF file.")
+    public File INPUT;
+
+    @Option(shortName=StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc="The output VCF/BCF file.")
+    public File OUTPUT;
+
+    @Option(shortName="F", doc="True if we are to fix the header, false if we are to replace the header (requires HEADER).")
+    public boolean FIX_HEADER = true;
+
+    @Option(shortName="N", doc="Check only the first N records when searching for missing INFO and FORMAT fields.", optional=true)
+    public int CHECK_FIRST_N_RECORDS = -1;
+
+    @Option(shortName="H", doc="The replacement VCF header.  Omit to infer from the input.", optional=true)
```

Comment:
"Omit to infer from the input." is a bit cryptic. I would leave it out entirely -- the argument itself is fairly self-explanatory, especially once you've read the tool description.